hub/saml-proxy/src/main/java/uk/gov/ida/hub/samlproxy/exceptions/SamlProxyDuplicateRequestExceptionMapper.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.ida.hub.samlproxy.exceptions;
+
+import com.google.inject.Provider;
+import uk.gov.ida.common.ErrorStatusDto;
+import uk.gov.ida.common.ExceptionType;
+import uk.gov.ida.common.SessionId;
+import uk.gov.ida.eventsink.EventSinkMessageSender;
+import uk.gov.ida.saml.hub.exception.SamlDuplicateRequestIdException;
+import uk.gov.ida.shared.utils.logging.LevelLogger;
+import uk.gov.ida.shared.utils.logging.LevelLoggerFactory;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Response;
+import java.util.Optional;
+import java.util.UUID;
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static uk.gov.ida.common.ExceptionType.INVALID_SAML_DUPLICATE_REQUEST_ID;
+
+public class SamlProxyDuplicateRequestExceptionMapper extends AbstractContextExceptionMapper<SamlDuplicateRequestIdException> {
+
+    private EventSinkMessageSender eventSinkMessageSender;
+    private final LevelLogger levelLogger;
+
+    @Inject
+    public SamlProxyDuplicateRequestExceptionMapper(
+            Provider<HttpServletRequest> contextProvider,
+            EventSinkMessageSender eventSinkMessageSender,
+            LevelLoggerFactory<SamlProxyDuplicateRequestExceptionMapper> levelLoggerFactory) {
+        super(contextProvider);
+        this.eventSinkMessageSender = eventSinkMessageSender;
+        this.levelLogger = levelLoggerFactory.createLevelLogger(SamlProxyDuplicateRequestExceptionMapper.class);
+    }
+
+    @Override
+    protected Response handleException(SamlDuplicateRequestIdException exception) {
+        UUID errorId = UUID.randomUUID();
+
+        Optional<SessionId> sessionId = getSessionId();
+        if (sessionId.isPresent()) {
+            eventSinkMessageSender.audit(exception, errorId, sessionId.get());
+        } else {
+            eventSinkMessageSender.audit(exception, errorId, SessionId.NO_SESSION_CONTEXT_IN_ERROR);
```

Comment:
`eventSinkMessageSender.audit(` shouldn't be duplicated really if just the last parameter is different.

Reply:
A good point - I checked the API docs for Optional, and learnt about orElse, which does exactly what we need here. I've made the same change to the other 2 incidences of this.